functions/xml_parse.py

```diff
@@ -3,6 +3,15 @@
 import copy
 import logging
 
+logger = logging.getLogger(__name__)
```

Comment:
You do not need to do this in every file you want to log.

use `logger.getLogger("<NAME>")` to get the logger as it is a singleton as long as you use the same `<name>`